apps/testclient/views.py

```diff
@@ -106,6 +108,7 @@ def restart(request):
     return render(request, HOME_PAGE, context={"session_token": None})
 
 
+@waffle_switch('enable_testclient')
 def callback(request):
```

Comment:
Should we should add a safe guard so that the test client does not work with non-synthetic beneficiaries?

I was thinking of something like this code in the `callback(request)` function right after the token is fetched at line # 142 

```
    # Check if returned token is for a synthetic bene and error if not.
    if int(token.get("patient", "0")) > 0:
        logmsg = "Failed token is for a non-synthetic patient ID = %s" % (token.get("patient", ""))
        logger.error(logmsg)
        if 'token' in request.session:
            del request.session['token']

        return JsonResponse({'error': logmsg,
                             'code': 'NonSyntheticTokenError',
                             'help': 'Try authorizing again.'}, status=403)

```

Reply:
That's a good idea, I like the failsafe on this.

I think we can make this for ALL test clients, not just the prod one right?
I can't see any reason that sandbox requests should go to non-synthetic bene's.

Reply:
@JFU-GIT This code might be better to use for the conditional:

```
# Check if returned token is for a synthetic bene and error if not.
    if not patient.startswith("-"):
```

Reply:
like the safe guard, and it applies to all ENV.